Neos.Flow/Classes/Http/BaseRequest.php

```diff
@@ -0,0 +1,389 @@
+<?php
+namespace Neos\Flow\Http;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Neos\Utility\MediaTypes;
+
+/**
+ *
+ */
+class BaseRequest extends AbstractMessage implements RequestInterface
+{
+    /**
+     * @var Uri
+     */
+    protected $uri;
+
+    /**
+     * @var Uri
+     */
+    protected $baseUri;
+
+    /**
+     * @var string
+     */
+    protected $method = 'GET';
+
+    /**
+     * URI for the "input" stream wrapper which can be modified for testing purposes
+     *
+     * @var string
+     */
+    protected $inputStreamUri = null;
+
+    /**
+     * @var string
+     */
+    protected $requestTarget = null;
+
+    /**
+     * BaseRequest constructor.
+     *
+     * @param UriInterface $uri
+     * @param string $method
+     */
+    public function __construct(UriInterface $uri, $method = 'GET')
```

Comment:
You expect `UriInterface` in the constructor, but the doc comment for `$this->uri` states that it is a concrete `Uri`. 

Why does the constructor argument `$method` need to be optional? 

The `$method` argument does not have a `string` type hint. And should be checked if it actually is a string with length > 0.